### ISSUE TITLE
The redundancy settings in UI State sync file

### ISSUE THREAD

**Issue by YOYOYOYOGA-seu** [microsoft/vscode#159749]
There are too many WorkspaceTrust settings under the value "workbench.statusbar.hidden" in globalState.json, and can't be cleaned, even if you disable the workspace trust function.
I think a auto clean function is needed.
![image](https://user-images.githubusercontent.com/53963800/187896444-eac9cfc6-15a3-4dbe-a5c6-47845fb17998.png)

**Comment by bpasero** [comment]
From: https://github.com/microsoft/vscode/blob/a4dd83d69b931aa27030111eebc853b04ef716f0/src/vs/workbench/contrib/workspace/browser/workspace.contribution.ts#L207

**Comment by sbatten** [comment]
This is because we create a status bar item per workspace, we don't have control over that storage since status bar entries are global. Do we just switch to a single item and manage our own workspace state?

**Comment by bpasero** [comment]
Yes, I like the idea of a single status bar entry and having a separate storage field to remember the association with the workspace.

**Comment by sbatten** [comment]
This was addressed in https://github.com/microsoft/vscode/commit/1b58747581e (PR https://github.com/microsoft/vscode/pull/203557), which simplified the workspace trust status bar entry to use a single ID instead of creating one per workspace — shipped in VS Code 1.87. The redundant entries no longer accumulate in global state. Closing as completed.
